frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -217,6 +217,16 @@ export function InstantSearchProvider({
                   ? filter
                   : {};
 
+            Object.entries(filterObject).forEach(([attribute, value]) => {
+               const widgetType = getFacetWidgetType(attribute);
+               if (
+                  widgetType === FacetWidgetType.Menu &&
+                  Array.isArray(value)
+               ) {
+                  filterObject[attribute] = value[0];
+               }
+            });
```

Comment:
This does fix the really specific case of submitting an array-like value to a param that doesn't accept an array, but there are many more ways this could 500. I'm more inclined to validate each param and ignore it if it's not the right format rather than auto-correcting a specific failure mode.

Thoughts?

Reply:
see https://github.com/iFixit/react-commerce/pull/1057/commits/f7e1d7f4bee2053bd0ec82a6da7e3f482573774f and lmk what you think